### DISCUSSION TITLE
Is the Direct Play guide still relevant?

### DISCUSSION THREAD

**Discussion by roenthomas** [croneter/PlexKodiConnect#1254] [Q&A] [answered]
I can get direct play without following the guide steps in the wiki, are these steps still relevant in the current version?

https://github.com/croneter/PlexKodiConnect/wiki/Direct-Play

**Comment by croneter** [comment, marked as answer]
Yes that's still very relevant. You need to have direct paths configured to enjoy direct play.

**Reply by roenthomas** [reply]
What I mean is that I used the default addon setup without ever setting up direct paths, and I was able to direct play from my plex server.

**Comment by croneter** [comment]
That was probably never direct play but direct streaming. You cannot tell within Kodi or PKC, only from the Kodi log. But for me this is semantics really and shouldn't matter much...

The one exception is you using Kodi on the same PC where the Plex Media Server is installed. Then direct play works out of the box.

**Reply by roenthomas** [reply]
Interesting.

I use Tautulli to monitor my server activity and generally watch on a Vero 4K+ with the PC being the Plex Server. Tautulli, at least AFAIK, differentiates between Direct Play, Direct Stream and Transcoding for each stream. With the default setup and not having to mess with direct path setup, I've managed to find situations where all 3 (Direct Play, Direct Streaming and Transcoding) occur, depending on the media file being played.

I take it there are one of two explanations:
1. This is not intended behavior
or
2. Tautulli is lying to me about the stream type?

**Reply by croneter** [reply]
Have a look at the Kodi log file and look for the string "The playful is". I don't know about Tautulli.

**Reply by croneter** [reply]
@roenthomas is your question answered?

**Reply by Iceman248** [reply]
On this subject, I set myself up for Direct Play and went through the guide to do so and the video and audio shows the original from source and as far as I can tell, it seems to be Direct Playing, but in the Plex Dashboard, it doesn't say "Direct Play", just shows video and audio quality. I didn't know if since not an official Plex app, maybe it didn't display that verbiage or what.

**Reply by croneter** [reply]
Exactly, @Iceman248 , the official Plex dashboard does not know about direct paths nor direct playing (PKC Definition). Hence the playback report won't show that

**Comment by moserroger** [comment]
Hello,

I followed this tutorial concerning direct paths:
https://github.com/croneter/PlexKodiConnect/wiki/Set-up-Direct-Paths

I can watch my movies and listen to my music on plex. I can watch the movies in kodi but the music won't play.

What could have gone wrong? What would I have to change to allow the music to play as well?

Thank you.

Cheers,

moses

**Reply by croneter** [reply]
In the pkc settings under Replace Paths, check the 2 settings for MUSIC. Then reset the Kodi database under Advanced

**Reply by moserroger** [reply]
Hi croneter

Thanks for the quick reply.

I used the "substitute paths on the PMS" method.

Under replace paths -> MUSIC I find no entry, nor for MOVIES OR TV SHOWS for that matter. On the other hand the "Plex UNC path replace option" was active. Would I have to add all the path unter Plex paths manually? Or just MUSIC if the movies work?

Unfortunately I couldn't find the "reset the Kodi database unter Advanced". I switched the Kodi menu to advanced but didn't find an option to reset the database.

Thanks for your help!

moses

**Reply by moserroger** [reply]
Update:
I actually did deactivate "Plex UNC path replace", then restarted Kodi and everything seems to work fine.  What could be the reason?

Well no, it's not true. I just found out that some albums work and some dont. And for some most of the songs work except some. I just realized something else. The songs that are played dont seem to be fully scraped. The description only shows the album name but not the year and other infos.

What could be going on?

Thank you.

**Reply by croneter** [reply]
Kindly open an issue here on GitHub and share a DEBUG log file of a full resync after you've reset the Kodi database. Then another log of some file not playing when another of the same album does

**Reply by moserroger** [reply]
I will gladly do that.

But I am too new to Kodi to understand all you say.

I know how to do a log.

But how to full resync?, and how to reset the Kodi database? I had a look online but didn't find anything.

Thanks

**Reply by moserroger** [reply]
I had another look at the instruction page I meantioned before. I just realized that I misunderstood the instructions and just did the PMS part but not the PKC one.

I will complete the instructions and see what it turns out to be.

Sorry if I caused a lot of confusion because of a lack of understanding the instructions.

**Reply by moserroger** [reply]
Hi croneter

I have now finish the tutorial.

At the end it says:
"Under Advanced, reset the Kodi databases with the red Partial or full reset of Database and PKC"
How to reset?

**Reply by croneter** [reply]
Pkc settings -> Advanced -> Reset the Kodi database (in red) -> YES -> NO
